test/run_test.sh

```diff
@@ -8,57 +8,30 @@ run_test_report=run_test_report_${TS}.txt
 > $run_test_log
 > $run_test_report
 
-echo "running test_deflate..."
-$TEST_WRAPPER ./test_deflate >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_deflate failed."
-    exit 1;
-fi
-echo "running test_inflate..."
-$TEST_WRAPPER ./test_inflate >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_inflate failed."
-    exit 1;
-fi
-echo "running test_inflatesyncpoint..."
-$TEST_WRAPPER ./test_inflatesyncpoint >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_inflatesyncpoint failed."
-    exit 1;
-fi
-echo "running test_stress..."
-$TEST_WRAPPER ./test_stress  >> $run_test_log 2>&1
-echo "running test_crc32..."
-$TEST_WRAPPER ./test_crc32   >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_crc32 failed."
-    exit 1;
-fi
-echo "running test_adler32..."
-$TEST_WRAPPER ./test_adler32 >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_adler32 failed."
-    exit 1;
-fi
-echo "running test_multithread_stress..."
-nthreads=${TEST_NTHREADS:-$(nproc)}
-$TEST_WRAPPER ./test_multithread_stress $nthreads 10 6  >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_multithread_stress failed."
-    exit 1;
-fi
-echo "running test_pid_reuse..."
-$TEST_WRAPPER ./test_pid_reuse 2 >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_pid_reuse failed."
-    exit 1;
-fi
-echo "running test_zeroinput..."
-$TEST_WRAPPER ./test_zeroinput >> $run_test_log 2>&1
-if [ $? -ne 0 ]; then
-    echo "test_zeroinput failed."
-    exit 1;
-fi
+run() {
+    test="$1"
+    shift 1
+    testname=$(basename $test)
+
+    echo "running ${testname}..."
+    $TEST_WRAPPER $test "$@" >> $run_test_log 2>&1
+    if [ $? -ne 0 ]; then
+        echo "${testname} failed."
+        exit 1;
+    fi
+}
+
+run ./test_abi ../lib/libnxz.so
```

Comment:
Is this passing even when using `--drop-undefined-syms`?

Reply:
Not yet, I'm still trying to understand what is going on there. If we stop using `--drop-undefined-syms` when generating the `.abi` files, then the issue disappears.

Reply:
@tuliom Those were being caused by some incompatible function declarations for `crc32` and `nx_adler32_combine`. I added a new commit to address that, regenerated `libnxz.api`, and now all tests pass.